buildSrc/src/main/kotlin/io/spine/internal/gradle/git/UserInfo.kt

```diff
@@ -26,37 +26,25 @@
 
 package io.spine.internal.gradle.git
 
-import com.google.api.client.util.Preconditions
-
 /**
- * Wrapper for different Git settings.
+ * Contains information about a Git user.
+ *
+ * Determines the author and committer fields of a commit.
  */
-class Config {
+data class UserInfo private constructor(val name: String, val email: String) {
 
-    /**
-     * Encapsulates `user.name` and `user.email` settings. These settings determine
-     * what ends up in author and commiter fields of a commit.
-     */
-    data class User private constructor(val name: String, val email: String) {
-        companion object Factory{
-            /**
-             * Validates provided parameters and constructs a [User] object.
-             *
-             * @throws IllegalArgumentException if the name or the email is an empty
-             *         string.
-             */
-            fun of(name: String, email: String): User {
-                Preconditions.checkArgument(
-                    name.isNotBlank(),
-                    "Name cannot be an empty string."
-                )
-                Preconditions.checkArgument(
-                    email.isNotBlank(),
-                    "Email cannot be an empty string."
-                )
+    companion object Factory {
+        /**
+         * Validates provided parameters and constructs a [UserInfo] object.
+         *
+         * @throws IllegalArgumentException if the name or the email is an empty
+         *         string.
+         */
+        fun of(name: String, email: String): UserInfo {
```

Comment:
@tripolkaandrey, and why not have just a public constructor for `UserInfo`? 

We don't cache these objects, we don't do anything  special about them. If it's just for checking preconditions, then `init {}` section in Kotlin serves this for this purpose just fine.